### PR TITLE
Api integration for weather app

### PR DESCRIPTION
**Pull request by shubhams3423**
fix/search place problem and UI updated.
Removed unused packages.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,13 +3,18 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@fortawesome/free-solid-svg-icons": "^6.4.2",
```
Comment (sashaboi) [inline package.json:6]: do we need both react icons and font awesome? could we try to work with only one? 
Please try to keep to one library.

File: package.json [hunk 2]
```diff
@@ -3,13 +3,18 @@
...
+    "@fortawesome/react-fontawesome": "^0.2.0",
     "@testing-library/jest-dom": "^5.17.0",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-icons": "^4.10.1",
+    "react-loading-skeleton": "^3.3.1",
```
Comment (sashaboi) [inline package.json]: are you using skeleton somewhere?

File: src/App.js [hunk 3]
```diff
@@ -2,6 +2,7 @@
 import './App.css';
 import WeatherApp from './components/WeatherApp';
 
+
```
Comment (sashaboi) [inline src/App.js:5]: unnecessary line break, please remove

File: src/components/WeatherApp.jsx [hunk 4]
```diff
@@ -1,56 +1,110 @@
...
+import 'react-toastify/dist/ReactToastify.css';
 import { FiSearch } from 'react-icons/fi'
-import { BsDropletFill } from 'react-icons/bs'
 import { FaWind } from 'react-icons/fa'
 const WeatherApp = () => {
+    const [weatherData, setWeatherData] = useState({})
+    const [isLoading, setIsLoading] = useState(false)
+    const [inputName, setInputName] = useState("Pune");
+    const [saveInputText, setSaveInputText] = useState("Pune");
+    const [error, setError] = useState({ errorText: "", showError: "removeErrorSymbol" })
```
Comment (sashaboi) [inline src/components/WeatherApp.jsx]: Why do we need this default error state, what does it do?

File: src/components/WeatherApp.jsx [hunk 5]
```diff
@@ -1,56 +1,110 @@
...
+                    setError({ errorText: "?", showError: "displayErrorSymbol" })
+                    toast.error("Location in valid , please enter a city name.")
+                }
+                else {
+                    setWeatherData(data);
+                    setError({ errorText: "", showError: "removeErrorSymbol" })
+                }
+
+            })
             .catch((error) => console.log(error))
```
Comment (sashaboi) [inline src/components/WeatherApp.jsx]: We need to put error here as well. toastify. 
Above error is for when api is successful but what if weatherapi itself is broken or down? 
We need to handle for that. 
Add toast for catch as well please. 
Try to make a common function in some new helper file. 
And call it in your data.error and in catch. 
let me know if you have any questions about this.

File: src/components/WeatherApp.jsx [hunk 6]
```diff
@@ -1,56 +1,110 @@
...
-    }, [])
+    }, [saveInputText])
+    const checkOnKeyDown = (e) => {
+        if (e.key === "Enter") {
```
Comment (sashaboi) [inline src/components/WeatherApp.jsx:34]: nice touch 👍

File: src/components/WeatherApp.jsx [hunk 7]
```diff
@@ -1,56 +1,110 @@
...
+                else {
+                    setWeatherData(data);
+                    setError({ errorText: "", showError: "removeErrorSymbol" })
+                }
+
+            })
             .catch((error) => console.log(error))
-    }, [])
+    }, [saveInputText])
+    const checkOnKeyDown = (e) => {
```
Comment (sashaboi) [inline src/components/WeatherApp.jsx]: lets use function naming like this : `handle`+`FunctionPurpose`
so above function will be named `handleInputBoxKeydown`

File: src/components/WeatherApp.jsx [hunk 8]
```diff
@@ -1,56 +1,110 @@
...
                     <div className="searchSection">
-                        <input type="text" className='inputText' placeholder='Pune' />
+                        <input type="text" className='inputText' placeholder='Search...' value={inputName} onChange={(e) => setInputName(e.target.value)} onKeyDown={checkOnKeyDown} />
                         <div className='searchIcon'>
-                            <FiSearch size={19} />
+                            <FiSearch onClick={onClickRenderWeather} />
                         </div>
                     </div>
                     <div >
                         <img src="https://previews.123rf.com/images/martialred/martialred1702/martialred170200035/71710170-cloudy-or-cloud-partly-blocking-the-sun-flat-color-vector-icon-for-weather-apps-and-websites.jpg" alt="weather Img" className=' weatherImg' />
```
Comment (sashaboi) [inline src/components/WeatherApp.jsx:53]: We need to work on this next pr

File: src/components/WeatherApp.jsx [hunk 9]
```diff
@@ -1,56 +1,110 @@
...
                     </div>
-                    <div>
-                        <div className="temperature ">
-                            <h2>28 &#8451;</h2>
-                            <p>Scattered Clouds</p>
+                    <div  >
```
Comment (sashaboi) [inline src/components/WeatherApp.jsx]: spacing in div tag remove. ( dont you have a linter installed? )

File: src/components/WeatherApp.jsx [hunk 10]
```diff
@@ -1,56 +1,110 @@
...
+                        <div className='subMainSection'>
+                            <div className="tempMainDiv ">
+                                <div className='tempSubDiv'>
+                                    <h2 className={`h2-${error.showError}`}>{isLoading ? <BeatLoader
```
Comment (sashaboi) [inline src/components/WeatherApp.jsx]: i didnt understand how is this error.showError working? 
why not do it like
className = { error ? "h2-YourErrorClassname" : "h2 } 
much more readable right?

File: src/components/WeatherApp.jsx [hunk 11]
```diff
@@ -1,56 +1,110 @@
...
+
                         </div>
                         <div className='lastSection'>
                             <div className="weatherSection">
                                 <div>
-                                    <BsDropletFill size={31} />
+                                    {isLoading ? <BeatLoader
+                                        color="#252827"
+                                        size={10}
+                                    /> : weatherData.error ? error.errorText : weatherData?.current?.humidity <= 55 ? <FontAwesomeIcon icon={faDroplet} className='drop' style={{ color: "rgba(23, 91, 208, 0.42)", }} /> : (weatherData?.current?.humidity <= 65 && weatherData?.current?.humidity > 55) ? <FontAwesomeIcon icon={faDroplet} className='drop' style={{ color: "rgb(86, 145, 247)", }} /> : <FontAwesomeIcon icon={faDroplet} className='drop' style={{ color: "rgb(23, 91, 208)", }} />}
```
Comment (sashaboi) [inline src/components/WeatherApp.jsx]: All this in one line is not readable please split in multiple lines.

**Review by sashaboi — CHANGES_REQUESTED** [review]
Take a day or two to work on these changes, let me know if you have any questions. 
Go one by one, dont be scared by number of comments. all are easy and doable.

File: src/components/WeatherApp.jsx (hunk 10 above)
Comment (shubhams3423) [reply src/components/WeatherApp.jsx]: yes.

File: package.json (hunk 2 above)
Comment (shubhams3423) [reply package.json]: No I removed it.

File: src/components/WeatherApp.jsx (hunk 4 above)
Comment (shubhams3423) [reply src/components/WeatherApp.jsx]: yes I understood. No need of error state.

**Review by sashaboi — APPROVED** [review]

**Pull request merged**